app/views/artisans/ArtisanRootView.coffee

```diff
@@ -0,0 +1,63 @@
+RootView = require 'views/core/RootView'
+
+Campaigns = require 'collections/Campaigns'
+Levels = require 'collections/Levels'
+
+module.exports = class ArtisanRootView extends RootView
+  id: 'artisan-root-view'
+  excludedCampaigns = [
+    # Misc. campaigns
+    'picoctf', 'auditions'
+
+    # Campaign-version campaigns
+    #'dungeon', 'forest', 'desert', 'mountain', 'glacier'
+
+    # Test campaigns
+    'dungeon-branching-test', 'forest-branching-test', 'desert-branching-test'
+
+    # Course-version campaigns
+    #'intro', 'course-2', 'course-3', 'course-4', 'course-5', 'course-6'
+  ]
+
+  unloadedCampaigns: 0
+  campaignLevels: {}
+  loadedLevels: {}
+  data: {}
+  problemCount: 0
+
+  initialize: ->
+    @campaigns = new Campaigns([])
+    @listenTo(@campaigns, 'sync', @onCampaignsLoaded)
+    @supermodel.trackRequest(@campaigns.fetch(
+      data:
+        project:'slug'
+    ))
+
+  onCampaignsLoaded: (campCollection) ->
+    for campaign in campCollection.models
+      campaignSlug = campaign.get('slug')
+      continue if campaignSlug in excludedCampaigns
+      @unloadedCampaigns++
+
+      @campaignLevels[campaignSlug] = new Levels()
+      @listenTo(@campaignLevels[campaignSlug], 'sync', @onLevelsLoaded.bind @, campaignSlug)
+      @supermodel.trackRequest(@campaignLevels[campaignSlug].fetchForCampaign(campaignSlug,
+        data:
+          project: 'thangs,name,slug,campaign,tasks'
+      ))
+
+  onLevelsLoaded: (campaignSlug, lvlCollection) ->
+    for level, k in lvlCollection.models
+      level.campaign = campaignSlug
+      @loadedLevels[campaignSlug] = {} unless @loadedLevels[campaignSlug]?
+      ll = {} unless ll?
+      level.seqNo = lvlCollection.models.length - k
+      @loadedLevels[campaignSlug][level.get('slug')] = level
+    if --@unloadedCampaigns is 0
+      @onAllLevelsLoaded?()
+
+  onAllLevelsLoaded: ->
+    console.log @render, @loadedLevels
+    @render()
```

Comment:
All this loading logic would make more sense in `Campaigns.coffee` so any view can take advantage of it without needing to be a certain subclass. Thinking something like this:

```
fetchCampaignsAndLevelsForArtisans: ->
  return Promise.resolve(@fetch())
  .then =>
    jqxhrs = []
    # ... make all your requests and push them onto `jqxhrs`
    return $when(jqxhrs...)
  .then =>
    # logic to tie them all together
```

And then in any view you can use

```
@campaigns = new Campaigns()
@supermodel.trackRequest @campaigns.fetchCampaignsAndLevelsForArtisans()
```